javaparser-core/src/main/java/com/github/javaparser/ast/expr/InstanceOfExpr.java

```diff
@@ -37,10 +37,13 @@
 import java.util.function.Consumer;
 import java.util.Optional;
 import com.github.javaparser.ast.Generated;
+import com.github.javaparser.metamodel.OptionalProperty;
 
 /**
  * Usage of the instanceof operator.
  * <br>{@code tool instanceof Drill}
+ * <br>JEP305 introduced the ability to assign a name to a matched node.
```

Comment:
If you take a look at other Javadocs, you'll see a structure:
1. general description
2. java version
3. specifics for the java version
4. small samples
5. an explanation how the parts of the samples map to the properties of the node
6. goto 2 :-)

I don't think JEP-references are in there, but they seem nice to add. I bet users think in Java versions, not in JEPs :-)

Reply:
The javadocs were just a dumping ground of "stuff" with very little thought given to structure ;)

definitely needs some fleshing out and polishing!! 

JEPs are helpful I think, with something like `(since JDK14, preview since JDK13)` or whatever is applicable in each case.

Reply:
That's cool. Here are the latest and pretty extensive Javadocs that I wrote for inspiration: https://www.javadoc.io/doc/com.github.javaparser/javaparser-core/latest/com/github/javaparser/ast/expr/SwitchExpr.html https://www.javadoc.io/doc/com.github.javaparser/javaparser-core/latest/com/github/javaparser/ast/stmt/SwitchEntry.html

Reply:
Those are rather nice docs! :1st_place_medal: